### ISSUE TITLE
Bug AddMonster command

### ISSUE THREAD

**Issue by Bari77** [OpenNos/OpenNos#1969]
Hi,
when i use AddMonster command, i set an existing MonsterVNum, he put him IG correctly, but in base, he put ID 0, and when i use again the command, he said me to use correctly and monster has spawn, but nothink (I guess he found an existing mob with ID 0).
I have try to change DB Colunm MapMonsterId to have Auto Increment, but GenericDAO InsertOrUpdate method dislike it :x.
Have you an idea ? This command work for u ? I have check last version of InsertOrUpdate method of GitHub, but same, and i have same DB as you.
PS : Another problem => I guess the caracter on the map not correctly instantiate, I have weapon on me, if I hit a monster, he took damage like I havn't weapon. Example : With weapon loaded, I hit 44k on mob, if I change map and come back, I hit again same mob, and he took 12k damage.
Bari.

**Comment by 0Lucifer0** [comment]
Seems logical. Would need a fix. But this is for issue not for question.

**Comment by Bari77** [comment]
Oh sorry for question then. Have said PS :
Another problem => I guess the caracter on the map not correctly instantiate, I have weapon on me, if I hit a monster, he took damage like I havn't weapon. Example : With weapon loaded, I hit 44k on mob, if I change map and come back, I hit again same mob, and he took 12k damage.

**Comment by 0Lucifer0** [comment]
damage are calculated on hit map hasn’t anything to do with it. Seems strange.

**Comment by Bari77** [comment]
I try somethink and i detail the problem and all cases.

**Comment by Bari77** [comment]
Have try many thinks, the problem is when you load character, if you look at character information, you see any stuff on you :
http://www.hostingpics.net/viewer.php?id=505124testproblem.png
And i guess, the monsters is binded on this. I'll try to fixe it, it's probably on a CharacterInstantiate cs file.
Thanks for help.

**Comment by 0Lucifer0** [comment]
That’s because your stuff are instantiate as iteminstance instead of wearable that was an error fixed some days ago just remove your iteminstances because they are corrupted.

**Comment by Bari77** [comment]
Perfect, thats work, thx.
For the initial bug (AddMonster Command), i guess it's because : https://github.com/OpenNos/OpenNos/commit/258221249d99dcd19f31247577bf9d868ed38172
You erase the line where you set the id.

**Comment by 0Lucifer0** [comment]
Yes this is the reason but the old line cant work... in fact I was thinking that mapmonster was auto incremented but it isn’t because of the parser. So you need to get a real number not from map but from all the server. Should be easy to fix. Don’t hesitate to pull request if you do it.

**Comment by Bari77** [comment]
Hum,
I'm new on this emulator, I havn't seen all files x), i'll search for fixe it, if i can i pull request you.
Bye.

**Comment by Bari77** [comment]
Can you give me your Discord name ? Or add [BFG] Bari#8834, i have some question for u.

**Comment by Bari77** [comment]
Have found a solution, I give you it :
The idea is to add "Last()" generic method on GenericDAO, this method need to return the last row of a dbset. When we get the last row of MapMonster table, we just set the MonsterId to row.id + 1.
So I had added following on DAL.EF/GenericDAO :
public TDTO Last()
        {
            using (OpenNosContext context = DataAccessHelper.CreateContext())
            {
                DbSet<TEntity> dbset = context.Set<TEntity>();
                return Mapper.Map<TDTO>(dbset.ToList().Last());
            }
        }
And I had changed the "GetNextMonsterId" method by following :
public int GetNextMonsterId()
        {
            //int nextId = _mapMonsterIds.Any() ? _mapMonsterIds.Last() + 1 : 1;
            int nextId = _mapMonsterIds.Any() ? DAOFactory.MapMonsterDAO.Last().MapMonsterId + 1 : 1;
            _mapMonsterIds.Add(nextId);
            return nextId;
        }
Closed.

**Comment by 0Lucifer0** [comment]
This is a bad idea because you access the database for nothing it doesn’t need database access. In addition keep issue open if there is no push to fix it. That’s avoid someone to open again
